samples/simpleasset/simpleasset.go

```diff
@@ -0,0 +1,238 @@
+	MaturityDate 	 time.Time    `json:"maturitydate"`
+}
+
+
+// InitLedger adds a base set of assets to the ledger
+func (s *SmartContract) InitLedger(ctx contractapi.TransactionContextInterface) error {
+	assets := []Asset{
+		{ID: "a1", AssetType: "type1", Issuer: "Treasury" , Owner: "", FaceValue: 300,
+			 MaturityDate: time.Date(2022, time.April, 1, 12, 0, 0, 0, time.UTC)},
+		{ID: "a2", AssetType: "type1", Issuer: "Treasury" , Owner: "", FaceValue: 400,
+			 MaturityDate: time.Date(2022, time.July, 1, 12, 0, 0, 0, time.UTC)},
+		{ID: "a3", AssetType: "type2", Issuer: "Treasury" , Owner: "", FaceValue: 1000,
+			 MaturityDate: time.Date(2023, time.April, 1, 12, 0, 0, 0, time.UTC) },
+		{ID: "a4", AssetType: "type3", Issuer: "Treasury" , Owner: "", FaceValue: 2000,
+			 MaturityDate: time.Date(2025, time.April, 1, 12, 0, 0, 0, time.UTC) },
+	}
+
+	for _, asset := range assets {
+		assetJSON, err := json.Marshal(asset)
+		if err != nil {
+			return err
+		}
+
+		err = ctx.GetStub().PutState(asset.ID, assetJSON)
+		if err != nil {
+			return fmt.Errorf("failed to put to world state. %v", err)
+		}
+	}
+
+	return nil
+}
+
+// CreateAsset issues a new asset to the world state with given details.
+func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface, id string, assettype string, owner string, issuer string, faceValue int, maturityDate time.Time) error {
+	exists, err := s.isAssetExists(ctx, id)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return fmt.Errorf("the asset %s already exists", id)
+	}
+
+	asset := Asset{
+		ID:             id,
+		AssetType:           assettype,
+		Owner:          owner,
+		Issuer:         issuer,
+		FaceValue: 		faceValue,
+		MaturityDate: 	maturityDate,
+	}
+	assetJSON, err := json.Marshal(asset)
+	if err != nil {
+		return err
+	}
+
+	return ctx.GetStub().PutState(id, assetJSON)
+}
+
+// ReadAsset returns the asset stored in the world state with given id.
+func (s *SmartContract) ReadAsset(ctx contractapi.TransactionContextInterface, id string) (*Asset, error) {
+	assetJSON, err := ctx.GetStub().GetState(id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read from world state: %v", err)
+	}
+	if assetJSON == nil {
+		return nil, fmt.Errorf("the asset %s does not exist", id)
+	}
+
+	var asset Asset
+	err = json.Unmarshal(assetJSON, &asset)
+	if err != nil {
+		return nil, err
+	}
+
+	return &asset, nil
+}
+
+// DeleteAsset deletes an given asset from the world state.
+func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface, id string) error {
+	exists, err := s.isAssetExists(ctx, id)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("the asset %s does not exist", id)
+	}
+
+	return ctx.GetStub().DelState(id)
+}
+
+// AssetExists returns true when asset with given ID exists in world state
+func (s *SmartContract) isAssetExists(ctx contractapi.TransactionContextInterface, id string) (bool, error) {
```

Comment:
Perhaps change the function name to `doesAssetExist` for grammatical correctness?

Reply:
Also, for consistency, can we have all function names representing chaincode transactions begin with a capital letter? I checked other chaincode modules in the repository and that's the naming scheme we've been using.

Reply:
Okay, I too thought about `doesAssetExist`, but it seemed strange as a function name, what about just `AssetExists`?
`if (AssetExists)` would be better programmatically right?

Reply:
Sure, `AssetExists` is fine.